C/services/common/include/plugin_api.h

```diff
@@ -16,7 +16,8 @@ typedef struct {
         unsigned int	options;
         const char	*type;
         const char	*interface;
-	const char	*config;
+	    const char	*config;
+	    bool        deprecated;
```

Comment:
This was not meant to change the plugin_info structure. The Jira was to add a new flag value to the flags bit mask.

Reply:
What will be the value for flags bit mask? And what about for python plugins as there is NO Flag key under plugin_info routine.

Reply:
Starting with the Python side. Here the approach you have taken is fine as it is a dict and inherently flexible.

On the C side, the current defined flag bits are
#define SP_COMMON       0x0001
#define SP_READINGS     0x0002
#define SP_ASYNC        0x0004
#define SP_PERSIST_DATA 0x0008
#define SP_INGEST       0x0010

in the file C/services/common/include/plugin_api.h

I would suggest we use the bit 

#define SP_DEPRECATED 0x0020